Cython/Compiler/ModuleNode.py

```diff
@@ -3239,6 +3241,7 @@ def generate_module_creation_code(self, env, code):
         else:
             doc = "0"
 
+        module_temp = code.funcstate.allocate_temp(py_object_type, manage_ref=True)
```

Comment:
The temp is never freed.

I think there should be some guard in Cython's infrastructure against this case that should just bark loudly.

Reply:
We need the extra reference because we return the module from the init function. I can go back to code that looks more like my earlier attempt where I let the module_temp live longer and use it in the return.

Reply:
Actually the code is simplest as is. Anyway is there anyway to make it more obvious to the reader that the ownership of the module is transfered to the returning code via return and stored in the meantime stored in `Naming.module_cname`?